framework/include/utils/NestedSolve.h

```diff
@@ -355,3 +367,123 @@ NestedSolve::nonlinear(V & guess, T compute)
   if (!is_converged())
     _state = State::NOT_CONVERGED;
 }
+
+template <typename V, typename B>
+bool
+NestedSolve::condition(
+    V & guess, B ci_lower_bounds, B ci_upper_bounds, unsigned int _num_eta, unsigned int _num_c)
+{
+  // check independent ci
+  for (unsigned int i = 0; i < _num_eta * _num_c; ++i)
+  {
+    if (guess[i] < ci_lower_bounds[i] || guess[i] == ci_lower_bounds[i] ||
+        guess[i] > ci_upper_bounds[i] || guess[i] == ci_upper_bounds[i])
+      return false;
+  }
+
+  // check dependent ci
+  for (unsigned int m = 0; m < _num_eta; ++m)
+  {
+    Real _sum_independent = 0;
+    for (unsigned int n = 0; n < _num_c; ++n)
+      _sum_independent += guess[n * _num_eta + m];
+
+    if (_sum_independent > 1 || _sum_independent == 1 || _sum_independent < 0 ||
+        _sum_independent == 0)
+      return false;
+  }
+  return true;
+}
+
+template <typename V, typename T, typename B>
+void
+NestedSolve::nonlinear(V & guess,
```

Comment:
it should be possible to not make this a separate function. I see a bit of code duplication here.

Reply:
I made condition() to be a lambda function inside NestedSolve::nonlinear().